content/blog/2022/2022-08-25-schedule-cron-job-in-node.md

```diff
@@ -0,0 +1,277 @@
+ # │ │ │ │ │ │
+ # *  * * * * *  command to execute
+```
+You can replace each asterisk with an appropriate number (or character) such that the expression describe the time the job will be executed.
+
+The first field (second) is optional. Each field can have one or more values separated by commas, or a range of values separated by hyphens.
+
+- `*` Asterisk operator this is a wildcard and acts as a placeholder. It means a value or always. Eg. If the asterisk symbol is in the month field, it means performing these specified tasks every month.
+- `,` The comma operator allows you to specify a list of values for repetition. For example, If you have `1, 3, 5` in the `month field`, the task will run on days of `month 1, 3, and 5`.
+- `-` The hyphen operator allows you to specify a range of values. If you have 1-5 in the `Day of the week field`, the task will run every weekday (From Monday to Friday).
+- `/` The slash operator allows you to specify values that will be repeated over a certain interval between them. For example, if you have */4 in the Hour field, It means the action will be performed every 4 hours. 
+
+A single asterisk means the specified job will run for every instance of that unit of time it represents. 
+
+Eg. cron expressions are started with five asterisks `* * * * *` by default this means: 
+
+Perform the specified task `at every minute`, `of every hour`, `of every day`, `of every month`, `and of every day of the week`.
+
+If you're unsure about manually writing cron syntax, you can, use free tools like [Crontab Generator](https://crontab-generator.org/) or [Crontab.guru](https://crontab.guru/#*_*_*_*_*) to generate the exact time date you want for cron expression.
+
+- **The second argument** specifies a function. This is the task that executes at intervals.  When the time specified in the first argument is reached, this function is called.
+
+- **The third argument** is a configuration object for the scheduled jobs, which is optional.
+
+#### The available options are:
+- scheduled: boolean to indicate whether the task created is scheduled (default is true).
+- timezone: because timezones differ depending on location, this option provides consistency in scheduled jobs by forcing them to adhere to a specific timezone or area.
+
+## Setting up your Node.Js application
+Installing dependencies and creating a node application.
+
+To begin, open your terminal and create a new folder.
+```
+mkdir node-cron-demo
+```
+
+Next change into the new project's directory
+```
+cd node-cron-demo
+```
+You will need to create a file index.js here, this is where we'll be writing all our codes.
+```
+touch index.js
+```
+Now run the command below to initialize the project. This will generate a `package.json` file which you can use to keep track of all project dependencies.
+```
+npm init -y
+```
+
+Next, let's add node-cron and other modules used later in this post.
+```
+npm install node-cron node-mailer 
+```
+
+## What we will be building
```

Comment:
I would rather name this section something like "Implementing Scheduling Use Cases" or similar. The section header "What we will be building" doesn't make sense when you look at the table of contents:

<img width="334" alt="image" src="https://user-images.githubusercontent.com/1363578/187295502-255a4fb4-d421-40af-8e26-7d35d80bc586.png">